src/extension/background/vault-manager.ts

```diff
@@ -44,27 +48,40 @@ let vault: Vault = {
   ...defaultVault,
   encryptedSecretKey: localStorage.getItem(encryptedKeyIdentifier) || undefined,
   hasSetPassword: getHasSetPassword(),
+  salt: localStorage.getItem(saltIdentifier) || undefined,
 };
 
 export const getVault = () => {
   return vault;
 };
 
-export const vaultMessageHandler = async (message: MessageFromApp) => {
-  vault = await vaultReducer(message);
-  if (vault.encryptedSecretKey) {
-    localStorage.setItem(encryptedKeyIdentifier, vault.encryptedSecretKey);
+function persistOptional(storageKey: string, value?: string) {
+  if (value) {
+    localStorage.setItem(storageKey, value);
   } else {
-    localStorage.removeItem(encryptedKeyIdentifier);
+    localStorage.removeItem(storageKey);
   }
+}
+
+export const vaultMessageHandler = async (message: MessageFromApp) => {
```

Comment:
```suggestion
export async function vaultMessageHandler(message: MessageFromApp) {
```

To be consistent with others in this file.